debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -300,6 +312,55 @@ protected void readChunk() throws InterruptedException {
         }
     }
 
+    private boolean isTableInvalid() {
+        final TableId currentTableId = (TableId) context.currentDataCollectionId();
+        currentTable = databaseSchema.tableFor(currentTableId);
+        if (currentTable == null) {
+            LOGGER.warn("Schema not found for table '{}', known tables {}", currentTableId, databaseSchema.tableIds());
+            nextDataCollection();
+            return true;
+        }
+        if (currentTable.primaryKeyColumns().isEmpty()) {
+            LOGGER.warn("Incremental snapshot for table '{}' skipped cause the table has no primary keys", currentTableId);
+            nextDataCollection();
+            return true;
+        }
+        return false;
+    }
+
+    private void verifySchema() throws SQLException {
```

Comment:
Generally I understand the purpose and the flow of the method but it seems to me that's not easily readable for those who will go through it for the first time. Either code srestructure is needed or more comments are necessary.

Reply:
Refactored and added the comments